CONTRIBUTING.md

```diff
@@ -0,0 +1,184 @@
+Each change submitted in a pull request that addresses a different scope of the
+project should be contained within its own commit (with the exception of global
+changes, such as renaming files, that affect the entire scope of the project).
+
+For example, a pull request submitted for a new hardware component could
+include the following four commits:
+
+1. The first commit should add the headers and source files for the generic
+   driver class, if one is being added. (e.g. `drivers/<class>/<class>.c` and
+   `include/drivers/<class>.h`).
+2. The second commit should add the driver and its header
+   (e.g.  `drivers/<class>/<driver>.c` and `include/drivers/<class>/<driver>.h`).
+3. The third set of commits (one for each platform) should add the devices
+   that use this driver to `platform/<platform>/devices.c`. If this driver
+   supplies an additional platform-specific header, that should go in these
+   commits.
+4. Finally, the device should be utilized be used from common code (i.e. in
+   `common/main.c`).
+
+Changes to submitted pull requests that address feedback from project
+maintainers should be squashed into existing commits rather than being included
+at the end of a pull request as a new commit.
+
+### Formatting
+
+Crust firmware uses Uncrustify to format all source code files. Running `make
+format` will format all project files according to a predefined style guide.
+
+The following are _general_ formatting guidelines:
+- Lines should be no longer than 79 characters and a newline character.
+- Use tabs for indentation and spaces for alignment for all C source code and
+  header files.
+- Trim all trailing whitespace.
+- Use hyphens in strings, filenames, and documentation.
+- Use underscores in variable names and header guards.
+
+### C files
+
+Functions in drivers should be topologically sorted (i.e. the `probe` function
+at the bottom). Other functions should be sorted in alphabetical order with
+small, `static` helper functions at the top. Functions and macros in headers
+should be alphabetically sorted whenever possible.
+
+### Writing headers
+
+- Header files act as APIs and are reserved for functions that will be
+  _exposed_ to the rest of the project. Macros, `struct` definitions, and
+  function declarations that are only utilized by a driver or module belong in
+  the corresponding implementation file, rather than the header file.
+- Header file functions, prototypes, and declarations should be documented
+  according to [Doxygen JavaDoc style comment blocks][doxygen].
+- Functions defined in assembly should also be declared in a header, unless
+  they are not intended to be called directly (e.g. `__udivsi3`).
+- Ideally, the header should be named the same as the driver or source file
+  (except that `common/` and `lib/` should be omitted).
+- The header guard should be the same as the path you would need to put in your
+  `#include` directive, but with the following changes:
+   - All letters should be uppercase.
```

Comment:
One too many indent spaces

Reply:
This was meant to be rendered as a sub-list (and it does render correctly). Is there another way you would suggest to express this?

Reply:
You only need two spaces, not three. It should still render as a sublist

Reply:
Got it, thanks.